### PR TITLE
Refactor NodesStatsRequest to extract NodesStatsMetrics to a top-level class from it

### PR DESCRIPTION
**Pull request by NEUpanning**
This commit extracts `NodesStatsMetrics` to a top-level class from `NodesStatsRequest`. `NodesStatsMetrics` is a container that encapsulates the necessary information needed to indicate which node stats is requested.

Relates #100878

### CODE REVIEW

**Comment by NEUpanning** [comment]
@gmarouli Could you help review this PR? Thanks.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a container that encapsulates the necessary information needed to indicate which node stats is requested.
```
Comment (gmarouli) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java:24]: Shall we simplify this sentence a bit?

Something along the lines of, and this is just a suggestion:

> This class encapsulates the metrics and other information needed to define scope when we are requesting node stats. 

If the above makes sense.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ */
+public class NodesStatsMetrics implements Writeable {
```
Comment (gmarouli) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java:26]: Nit: Shall we rename this to match a bit more what it represents than which class is using it? What do you think?

**Review by gmarouli — CHANGES_REQUESTED** [review]
Thank you @NEUpanning for putting this together and for separating the concerns nicely in this PR.

It is looking really good, just a minor comment to see if we can find a better name, (naming is hard), and can you please resolve the conflict with main?

When this is done, I will trigger the tests.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java (hunk 2 above)
Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java:26]: How about `NodesStatsRequestParameters`? This name more accurately reflects the function and purpose of this class, that is, it is used to store the parameters of the node stats request.

File: docs/changelog/100883.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+pr: 100883
+summary: Refactor NodesStatsRequest to extract NodesStatsMetrics to a top-level class from it
+area: Stats
+type: refactoring
+issues: []
```
[the remark this answers is not in the document]
Comment (NEUpanning) [reply docs/changelog/100883.yaml]: I get it. I reverted this commit in 6c50f5b15a4183a3add463f9afc1199093d4c946.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java (hunk 1 above)
Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java]: Thanks. This suggeted sentence is more readable. It is finished in 7f0629e99f46d603d3b3d6ef18bc9fd5a9e0f96e

**Comment by NEUpanning** [comment]
The conflict with main is resolved in 74780eb9f88860355125f5e95be93b5d5f36dd81.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java (hunk 2 above)
Comment (gmarouli) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java:26]: I like it!

Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java]: I've finished it in 739ab8aec437c2a333d480a56a8919e1b8d9136b

**Comment by NEUpanning** [comment]
I think this PR is ready for testing :)

**Comment by gmarouli** [comment]
Can you fix the new conflicts and update with master before I trigger the test run?

**Comment by NEUpanning** [comment]
The new conflicts are fixed in 32f17743652e8611aef71bb30c2ac500b719ff43.

**Comment by gmarouli** [comment]
buildkite test this please

**Review by gmarouli — APPROVED** [review]
LGTM 🚀 ! Thank you @NEUpanning !

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
buildkite test this please

**Comment by NEUpanning** [comment]
@gmarouli Thank you for reviewing my code. Your feedback has been very helpful. I opened a new pull request #102559 for optimization. Could you take a look when you have some time? Thanks.

**Pull request merged**